segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ReadIndexCache.java

```diff
@@ -117,8 +117,12 @@ private void addIndexEntry(SegmentReadIndex segmentReadIndex, String streamSegme
                 .chunkName(chunkName)
                 .startOffset(startOffset)
                 .build();
-        segmentReadIndex.offsetToChunkNameIndex.put(startOffset, indexEntry);
-        indexEntryCache.put(indexEntry, true);
+        val existing = segmentReadIndex.offsetToChunkNameIndex.putIfAbsent(startOffset, indexEntry);
+        if (null == existing) {
+            indexEntryCache.put(indexEntry, true);
+        } else {
+            Preconditions.checkState(existing.equals(indexEntry), indexEntry.toString() + " != " + existing);
```

Comment:
more details?

Reply:
indexEntry.toString() prints out all the required details